engine/src/main/java/org/hibernate/search/engine/search/aggregation/spi/MetricDoubleAggregationBuilder.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.engine.search.aggregation.spi;
+
+import org.hibernate.search.engine.search.predicate.SearchPredicate;
+
+public interface MetricDoubleAggregationBuilder extends SearchAggregationBuilder<Double> {
+
+	interface TypeSelector {
+		MetricDoubleAggregationBuilder type();
+	}
```

Comment:
Note this "TypeSelector" step, as it is, is probably pointless. The only point of type selectors in other builders is to allow passing an expected type, e.g. like this:

https://github.com/hibernate/hibernate-search/blob/0b9e3f0d87250daa800f10223bcc81cdb0da773a/engine/src/main/java/org/hibernate/search/engine/search/aggregation/spi/TermsAggregationBuilder.java#L14-L16

I'm not saying you should remove this `TypeSelector` interface, but maybe there's something missing, e.g. a parameter to select the expected return type?
That's more or less what you're doing in `MetricFieldAggregationBuilder`...

Reply:
You're right. In the last commit I remove the `type()` methods to support non-fields aggregations. The code is simplified now.